bot.go

```diff
@@ -33,10 +34,31 @@ type BotAPI struct {
 // NewBotAPI creates a new BotAPI instance.
 //
 // It requires a token, provided by @BotFather on Telegram.
-func NewBotAPI(token string) (*BotAPI, error) {
+func NewBotAPI(token string, proxyparam ...interface{}) (*BotAPI, error) {
+	// create a socks5 dialer
+ 	if len(proxyparam) > 0 {
+ 		var auth *proxy.Auth
+		if proxyparam[0] == "socks5" {
+			if proxyparam[2] == nil {
+				auth = nil
+			} else {
+				auth = proxyparam[2].(*proxy.Auth)
+			}
+			dialer, err := proxy.SOCKS5("tcp", proxyparam[1].(string), auth, proxy.Direct)
+			if err != nil {
+				return nil, err
```

Comment:
I would suggest using `errors.Wrap(err, "telegram: explanation")` here. So that the returned error has more context.